### ISSUE TITLE
[enhancement] Handle damage values

### ISSUE THREAD

**Issue by Dablakbandit** [BrilliantTeam/Minecraft-ResourcePack-Migrator#3]
**Is your feature request related to a problem? Please describe.**
Resourcepacks still using damage values are not properly converted

**Describe the solution you'd like**
Use the "damage" property type and convert damage values to new format.

**Describe alternatives you've considered**
Cheap hack I've done, but probable a better way to do it:
https://github.com/AshleyThew/minecraft-resource-pack-upgrader/blob/3c9ce1801935b0382cec470e91b4437150607233/app/upgrade.py#L136
Unsure if we are able to mix n match damage values and custom_model_data in the new format

**Comment by RICE0707** [comment]
It seems like this is a parameter different from **custom model data**.  

Could you assist me in testing this on **1.21.4**? Please try using this JSON combining **custom model data** and **damage** to see if it works.
```
{
    "model": {
        "type": "range_dispatch",
        "property": "custom_model_data",
        "fallback": {
            "type": "model",
            "model": "minecraft:item/items/wood_sword"
        },
        "entries": [
            {
                "threshold": 2000,
                "model": {
                    "type": "range_dispatch",
                    "property": "damage",
                    "fallback": {
                        "type": "model",
                        "model": "minecraft:item/items/wood_sword"
                    },
                    "entries": [
                        {
                            "threshold": 0.25,
                            "model": {
                                "type": "model",
                                "model": "item/wooden_sword1"
                            }
                        },
                        {
                            "threshold": 0.5,
                            "model": {
                                "type": "model",
                                "model": "item/wooden_sword3"
                            }
                        },
                        {
                            "threshold": 0.75,
                            "model": {
                                "type": "model",
                                "model": "item/wooden_sword3"
                            }
                        }
                    ]
                }
            },
            {
                "threshold": 2001,
                "model": {
                    "type": "range_dispatch",
                    "property": "damage",
                    "fallback": {
                        "type": "model",
                        "model": "minecraft:item/items/wood_sword"
                    },
                    "entries": [
                        {
                            "threshold": 0.25,
                            "model": {
                                "type": "model",
                                "model": "item/wooden_sword4"
                            }
                        },
                        {
                            "threshold": 0.5,
                            "model": {
                                "type": "model",
                                "model": "item/wooden_sword5"
                            }
                        },
                        {
                            "threshold": 0.75,
                            "model": {
                                "type": "model",
                                "model": "item/wooden_sword6"
                            }
                        }
                    ]
                }
            }
        ]
    }
}
```

**Comment by RICE0707** [comment]
Got it! After further investigation, the method works perfectly.  

You just need to define both **custom model data** and **damage** values in the command, such as:  
`/give @s minecraft:diamond_hoe[minecraft:custom_model_data={floats:[2000]},minecraft:damage=781]`  

Additionally, I’ll provide an independent conversion mode to handle resource packs using only the **damage** parameter.  

Thank you for the information! I’ll release version **1.3** as soon as possible.

**Comment by andfont** [comment]
There are many shortcomings, many things are missing that it is not able to migrate, such as:
{
"__comment": "Max wood durability is 60",
"parent": "item/handheld",
"textures": {
"layer0": "minecraft:item/wooden_hoe"
},
"overrides": [
{"predicate": { "custom_model_data": 18001}, "model": "magic:wands/old_survival_wand1"},
{"predicate": {"damaged": 0, "damage": 0.01666666666667}, "model": "magic:wands/old_survival_wand1"},
{"predicate": {"damaged": 0, "damage": 0.03333333333333}, "model": "magic:wands/old_survival_wand2"},
{"predicate": {"damaged": 0, "damage": 0.05}, "model": "magic:wands/survival_wand1"},
{"predicate": {"damaged": 0, "damage": 0.06666666666667}, "model": "magic:wands/survival_wand2"},
{"predicate": {"damaged": 1, "damage": 0}, "model": "item/wooden_hoe"}
]
}
a combination of damage and layers etc.,..

**Comment by RICE0707** [comment]
Regarding **damage-based conversions**, since version **1.3** has not been released yet, the inability to perform such migrations is currently "expected behavior."

**Comment by andfont** [comment]
I understand, but is it possible in the meantime to show some manual example of how it would be? or how it would be done manually? thanks

**Comment by RICE0707** [comment]
For handling **custom model data + damage**, I recommend updating old resource packs to the following format:

```json
{
    "parent": "item/handheld",
    "textures": {
        "layer0": "items/wood_sword"
    },
    "overrides": [
        { "predicate": { "custom_model_data": 2000 }, "model": "item/wooden_sword0"},
        { "predicate": { "custom_model_data": 2000, "damaged": 1, "damage": 0.25 }, "model": "item/wooden_sword1"},
        { "predicate": { "custom_model_data": 2000, "damaged": 1, "damage": 0.50 }, "model": "item/wooden_sword2"},    
        { "predicate": { "custom_model_data": 2000, "damaged": 1, "damage": 0.75 }, "model": "item/wooden_sword3"},
        { "predicate": { "custom_model_data": 2001 }, "model": "item/wooden_sword00"},
        { "predicate": { "custom_model_data": 2001, "damaged": 1, "damage": 0.25 }, "model": "item/wooden_sword4"},
        { "predicate": { "custom_model_data": 2001, "damaged": 1, "damage": 0.50 }, "model": "item/wooden_sword5"},    
        { "predicate": { "custom_model_data": 2001, "damaged": 1, "damage": 0.75 }, "model": "item/wooden_sword6"}   
    ]
}
```

In the upcoming **1.3 version**, the following rules will apply:

1. If you choose the **damage mode**, **custom model data** will be ignored.
2. If you choose the **custom model data mode**, each `predicate` must include both `custom_model_data` and `damage` parameters to properly convert into the new format.

This ensures compatibility with the new conversion logic while maintaining flexibility for customized setups.

**Comment by RICE0707** [comment]
By the way, the current **1.2.4 version** has some deficiencies in the conversion logic, which will also be addressed in the upcoming **1.3 version**.

**Comment by RICE0707** [comment]
Version 1.3 has been released! Please update your old resource pack format according to the requirements mentioned above, and use the converter to upgrade it to the new format.
